instrumentation/opentelemetry-instrumentation-celery/src/opentelemetry/instrumentation/celery/utils.py

```diff
@@ -143,6 +153,64 @@ def detach_span(task, task_id, is_publish=False):
     span_dict.pop((task_id, is_publish), (None, None))
 
 
+def attach_span_context(
+    headers, span: trace.Span, parent_span: trace.Span
+) -> None:
+    if headers is None:
+        return
+
+    parent_ctx = parent_span.get_context()
+    ctx = span.get_context()
+
+    # We need to send objects as streams, as the amqp serializer appears to be unhappy
+    # when trying to serialize big numbers, like an uuid in integer form
+    headers[CTX_TRACE_ID] = str(parent_ctx.trace_id)
+    headers[CTX_PARENT_SPAN_ID] = str(parent_ctx.span_id)
+    headers[CTX_PUBLISHER_SPAN_ID] = str(ctx.span_id)
+    headers[CTX_PARENT_TRACE_FLAGS_SAMPLED] = getattr(
+        parent_ctx.trace_flags, "sampled", False,
+    )
```

Comment:
We should just use the configured propagator here to inject trace context and use propagator again to extract the context on the other side. No reason celery instrumentation should ignore globally configured propagators.